intents/swap/src/lib.rs

```diff
@@ -22,7 +24,7 @@ mod rollback;
 #[near(contract_state)]
 #[derive(PanicOnDefault)]
 pub struct SwapIntentContractImpl {
-    intents: LookupMap<IntentId, Mutex<SwapIntentStatus>>,
+    intents: LookupMap<IntentId, Mutex<SwapIntent>>,
```

Comment:
Does it make sense to change the `LookupMap` with the iterable `UnorderedMap` to allow to do garbage collection in the future? Also, such change will require making data migration, which is always fun 😄

Reply:
Good point, I'll fix it in a separate PR as we discussed